### PR TITLE
Assign Campaign to Contribution Pages for collection source

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Assign Campaign to Contribution Pages for collection source

## Summary by CodeRabbit

- **New Features**
	- Introduced a method to automatically update contributions with the correct campaign ID based on their collection source.

- **Bug Fixes**
	- Improved tracking of contributions to ensure they are linked to the appropriate campaigns.

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 1]
```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
     }
   }
 
+  /**
+   *
+   */
+  public static function updateCampaignForCollectionSourceContribution(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($objectName != 'Contribution' || $objectId === NULL) {
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1362]:
```suggestion
    if ($objectName != 'Contribution' || !$objectId) {
```

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 2]
```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
...
+      return;
+    }
+
+    $contributionId = $objectRef['id'] ?? NULL;
+    if (!$contributionId) {
+      return;
+    }
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1369]: I don't understand the need for this. This code is called on the pre hook so we can assume the contribution has not created just yet. Can't we?

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php (hunk 1 above)
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1362]: Can we also add a check on $op ? Either create or edit?

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 3]
```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
...
+
+    $contribution = Contribution::get(FALSE)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1371]: How are we getting the contribution? Are we running the code on create or edit?

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 4]
```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
...
+      ->addSelect('Contribution_Details.Source')
+      ->addWhere('id', '=', $contributionId)
+      ->execute()->first();
+
+    if (!$contribution) {
+      return;
+    }
+
+    $collectionSourceId = $contribution['Contribution_Details.Source'];
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1380]:
```suggestion
    $sourceID = $contribution['Contribution_Details.Source'];
```

**Review by pokhiii — COMMENTED** [review]
Some questions

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php (hunk 2 above)
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1369]: When I click on the contribute button after filling out the form, this hook runs. Inside the objectRef, I was able to get the contributionId. With the help of that contribution ID, I was fetching out the `collectionSourceId`. Based on that camp ID, I found the campaign associated with that camp.

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php (hunk 3 above)
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:1371]: I logged some data there where I was able to find the contribution id with the help of that I was able to find details

**Comment by tarunnjoshi** [comment]
this functionality is done on another PR https://github.com/ColoredCow/goonj/pull/601. Close with this comment

**Pull request closed without merging**